src/stcal/outlier_detection/utils.py

```diff
@@ -283,8 +289,36 @@ def calc_gwcs_pixmap(in_wcs, out_wcs, in_shape):
     bb = wcs_bbox_from_shape(in_shape)
     log.debug("Bounding box from data shape: {}".format(bb))
 
-    grid = gwcs.wcstools.grid_from_bounding_box(bb)
-    return np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
+    if fast and len(in_shape) == 2:
```

Comment:
Why is `len(in_shape) == 2` important here? How is the code different from the case of `fast=False`? That is, would the code handle more than 2 axes when `false=False`?